src/Processors/Formats/Impl/CSVRowInputFormat.cpp

```diff
@@ -25,10 +25,10 @@ namespace ErrorCodes
 
 namespace
 {
-    void checkBadDelimiter(char delimiter)
+    void checkBadDelimiter(char delimiter, bool use_whitespace_tab_as_delimiter)
     {
         constexpr std::string_view bad_delimiters = " \t\"'.UL";
-        if (bad_delimiters.find(delimiter) != std::string_view::npos)
+        if (bad_delimiters.find(delimiter) != std::string_view::npos && !use_whitespace_tab_as_delimiter)
```

Comment:
We should allow only space and tab when `use_whitespace_tab_as_delimiter=true`, not all bad delimiters

Reply:
done

Reply:
cc @Avogar